tests/test_constraint_manager.py

```diff
@@ -210,6 +210,148 @@ def test_multi_model_constraint_checks(self):
         self.assertFalse(
             ConstraintManager.satisfies_constraints(constraints, rcm))
 
+    def test_single_model_max_failure_percentage(self):
+        """
+        Test that constraint_failure_percentage works for a single model
+        with a max style constraint
+        """
+        config = self._create_single_model_with_constraints()
+        constraints = [
+            ConstraintManager.get_constraints_for_all_models(config)['model_A']
+        ]
+
+        # Constraint is P99 Latency max of 100
+        rcm = self._construct_rcm({"perf_latency_p99": 225})
+        self.assertEqual(
+            ConstraintManager.constraint_failure_percentage(constraints, rcm),
+            125)
+        rcm = self._construct_rcm({"perf_latency_p99": 150})
```

Comment:
Is there any concern that max-style constraints can be off by an infinite percentage, while min-style constraints are limited to 100%? I suppose that would only matter if there are BOTH min and max style constraints enabled at the same time?

Reply:
I'm not too concerned about that as I don't think that's a very common way of specifying constraints. My thoughts are the user will either want a minimum throughput or a max latency